components/formats-bsd/src/loci/formats/Memoizer.java

```diff
@@ -208,6 +208,17 @@ public void saveRevision(String revision) {
 
     @Override
     public void saveReader(IFormatReader reader) {
+      // clean up reader list in any instances of ImageReader
+      IFormatReader r = reader;
+      while (r instanceof ReaderWrapper || r instanceof ImageReader) {
```

Comment:
Suggest it would be cleaner to have ReaderWrapper and ImageReader both implement an interface 'ReaderContainer' (or use existing interface) for the method:
`IFormatReader getReader();`
Then test r for being an instance of that. Slightly bigger change so understand why not done :)

Since ImageReader requires the clean method the interface could have something like:
`IFormatReader getCleanReader();`

Which is implemented in ImageReader as `cleanupReaderList(); return getReader();` and in ReaderWrapper to just return `getReader()`.

Not a major point though.